lib/grape/endpoint.rb

```diff
@@ -210,47 +216,49 @@ def equals?(e)
     protected
 
     def run(env)
-      @env = env
-      @header = {}
+      ActiveSupport::Notifications.instrument('run.grape', endpoint: self, env: env) do
+        @env = env
```

Comment:
Fair enough, looks like the whole `run` method has potential for splitting up further.